kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -58,13 +58,9 @@ simplify
     => Pattern variable
     -> simplifier (OrPattern variable)
 simplify pattern' = do
-    simplifiedTerm <-
-        simplifyConditionalTermToOr predicate term
-    orPatterns <-
-        Branch.gather
-        $ traverse
-            (Pattern.mergeWithPredicate predicate)
-            simplifiedTerm
-    return (MultiOr.mergeAll orPatterns)
+    orSimplifiedTerms <- simplifyConditionalTermToOr predicate term
```

Comment:
I assume that this fixes the `or` evaluation bug you mentioned in the PR description. So, what was the bug? I guess I could figure it out if I keep staring at the change, but it's easier to just ask.l

Reply:
`traverse` distributes over `Or` like `And`, but we have an `Or` of `Or`; we want to concatenate, which `scatter` does.

The longer version is that `Branch` is hard to use correctly, and we should consider removing it.